### PR TITLE
Add Kotlin v3 docs

### PR DESCRIPTION
**Pull request by jan-tennert**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Docs update

## What is the new behavior?

New Kotlin v3 docs and some general improvements.
The most notable changes are the rename of `gotrue-kt` to `auth-kt`, storage uploading/downloading refactor and some breaking postgrest changes

### CODE REVIEW

File: apps/docs/docs/ref/kotlin/introduction.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@ This reference documents every object and method available in Supabase's Kotlin
 
 To see supported Kotlin targets, check the corresponding module README on [GitHub](https://github.com/supabase-community/supabase-kt).
 
-To migrate from version 1.4.X to 2.0.0, see the [migration guide](https://github.com/supabase-community/supabase-kt/blob/master/MIGRATION.md)
+To migrate from version 2.X to 3.0.0, see the [migration guide](https://github.com/supabase-community/supabase-kt/blob/master/MIGRATION.md)
```
[the remark this answers is not in the document]
Comment (jan-tennert) [reply apps/docs/docs/ref/kotlin/introduction.mdx:5]: Yea, that's temporary

**Review by charislam — APPROVED** [review]

**Comment by charislam** [comment]
@jan-tennert awesome! Let me know if/when this is ready to be published and I can merge it

**Comment by jan-tennert** [comment]
> @jan-tennert awesome! Let me know if/when this is ready to be published and I can merge it

:+1: Will do!

**Comment by jan-tennert** [comment]
@charislam Should be ready now

**Pull request merged**